samza-sql/src/main/java/org/apache/samza/sql/translator/ScanTranslator.java

```diff
@@ -74,9 +77,14 @@ void translate(final TableScan tableScan, final TranslatorContext context) {
     String sourceName = SqlIOConfig.getSourceFromSourceParts(tableNameParts);
 
     Validate.isTrue(relMsgConverters.containsKey(sourceName), String.format("Unknown source %s", sourceName));
-    final String streamName = systemStreamConfig.get(sourceName).getStreamName();
+    SqlIOConfig sqlIOConfig = systemStreamConfig.get(sourceName);
+    final String systemName = sqlIOConfig.getSystemName();
+    final String streamName = sqlIOConfig.getStreamName();
 
-    MessageStream<KV<Object, Object>> inputStream = streamGraph.getInputStream(streamName);
+    KVSerde<Object, Object> noOpKVSerde = KVSerde.of(new NoOpSerde<>(), new NoOpSerde<>());
```

Comment:
Same as above. Is this the final implementation?

Reply:
No, samza-sql changes TBD with Srini/Aditya.